seqr/views/apis/anvil_workspace_api.py

```diff
@@ -28,13 +28,27 @@
 
 anvil_auth_required = user_passes_test(is_anvil_authenticated, login_url=GOOGLE_LOGIN_REQUIRED_URL)
 
+BLOCK_SIZE = 65536
+
+
+def get_vcf_samples(vcf_filename):
+    byte_range = None if vcf_filename.endswith('.vcf') else (0, BLOCK_SIZE)
+    for line in file_iter(vcf_filename, byte_range=byte_range):
+        if line[0] != '#':
+            break
+        if line.startswith('#CHROM'):
+            return set(line.rstrip().split('FORMAT\t', 2)[1].split('\t') if line.endswith('\n') else [])
```

Comment:
I'm confused by the ` if line.endswith('\n')` check - what does having a newline have to do with anything here? Wouldn't a gzipped block end in the middle of a line some of the time?

Reply:
I think it is possible for a gzipped block to end in the middle of a line. If it happens then the `.endswith('\n')` checking can avoid using partial samples and let the user know something wrong.

Reply:
I mean, if there are partial samples then the missing sample check will fail and alert the user, so I think this is less helpful